lib/srv/desktop/audit.go

```diff
@@ -145,8 +149,188 @@ func (s *WindowsService) onClipboardReceive(ctx context.Context, emitter events.
 	s.emit(ctx, emitter, event)
 }
 
+// onSharedDirectoryAnnounce adds the shared directory's name to the auditCache.
+func (s *WindowsService) onSharedDirectoryAnnounce(sid string, m tdp.SharedDirectoryAnnounce) {
+	s.auditCache.SetName(sessionID(sid), directoryID(m.DirectoryID), directoryName(m.Name))
+}
+
+// onSharedDirectoryReadRequest adds ReadRequestInfo to the auditCache.
+func (s *WindowsService) onSharedDirectoryReadRequest(sid string, m tdp.SharedDirectoryReadRequest) {
+	s.auditCache.SetReadRequestInfo(sessionID(sid), completionID(m.CompletionID), readRequestInfo{
+		directoryID: directoryID(m.DirectoryID),
+		path:        m.Path,
+		offset:      m.Offset,
+	})
+}
+
+// onSharedDirectoryReadResponse emits a DesktopSharedDirectoryRead audit event.
+func (s *WindowsService) onSharedDirectoryReadResponse(
+	ctx context.Context,
+	emitter events.Emitter,
+	id *tlsca.Identity,
+	sid string,
+	desktopAddr string,
+	m tdp.SharedDirectoryReadResponse,
+) {
+	var did directoryID
+	var path string
+	var offset uint64
+	var name directoryName
+	// Gather info from the audit cache
+	info, ok := s.auditCache.GetReadRequestInfo(sessionID(sid), completionID(m.CompletionID))
+	if ok {
+		did = info.directoryID
+		// Only search for the directory name if we retrieved the directoryID from the audit cache.
+		name, ok = s.auditCache.GetName(sessionID(sid), did)
+		if !ok {
+			name = events.UnknownEvent
```

Comment:
All of these get changed to hardcoded `"unknown"`s further up the PR stack: https://github.com/gravitational/teleport/pull/17403/commits/d636f50f3f515b77be088393ba084b651b0d0bee